lib/rmt/file_validator.rb

```diff
@@ -0,0 +1,59 @@
+require 'fileutils'
+require 'rmt/checksum_verifier'
+
+class RMT::FileValidator
+  class << self
+    def validate_local_file(file_reference, deep_verify:)
+      verify_file_on_disk(file_reference, deep_verify).tap do |valid_on_disk|
+        update_file_status_on_database(file_reference, valid_on_disk)
+      end
+    end
+
+    def find_valid_file_by_checksum(checksum, checksum_type, deep_verify:)
+      ::DownloadedFile
+        .where(checksum: checksum, checksum_type: checksum_type)
+        .map { |file| [file, validate_tracked_file_on_disk(file, deep_verify)] }
+        .find { |_, valid_on_disk| valid_on_disk }
+        &.first&.local_path
+    end
+
+    private
+
+    def validate_tracked_file_on_disk(file_record, deep_verify)
```

Comment:
I'd consider renaming this method into `is_orphaned_db_record?`.

Reply:
Yes, the methods have become confusing with similar names like `validate_tracked_file_on_disk` and `verify_file_on_disk`. You can only really tell their meanings by looking at their code.

Reply:
`is_orphaned_db_record?` doesn't sounds good either, because the method tells if a file found on DB is valid on disk. Anyway, I rewrote the public method `#find_valid_file_by_checksum` and removed this method, as well as the complicate method chaining.